### PR TITLE
Update docs with the new Job annotation

### PR DESCRIPTION
**Pull request by helayoty**
Update the website docs with the new Job annotation that has been added as part of KEP https://github.com/kubernetes/enhancements/issues/4026.

### CODE REVIEW

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:1051]:
```suggestion
Used on: Jobs and Pods controlled by CronJobs
```

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:1053]:
```suggestion
This annotation is used to record the original (expected) creation timestamp for a Job,
when that Job is part of a CronJob.
The control plane sets the value to that timestamp in RFC3339 format.
```

Ideally, mention that the time zone comes from the CronJob when set.

**Review by sftim — COMMENTED** [review]
Thanks.

LGTM. There are also my suggested tweaks if you'd like 'em.

**Comment by sftim** [comment]
LGTM for docs

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (soltysh) [inline content/en/docs/reference/labels-annotations-taints/_index.md:1055]: I'd add a note about TimeZone, and how this affects this information.

Comment (helayoty) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: Addressed.

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (soltysh) [reply content/en/docs/reference/labels-annotations-taints/_index.md:1056]: We could, but I wouldn't want that, if you don't specify a TZ explicitly we will use kcm one.

**Review by soltysh — APPROVED** [review]
/lgtm
from sig-apps tech pov

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (soltysh) [reply content/en/docs/reference/labels-annotations-taints/_index.md:1056]: We can re-consider that before GA.

Comment (helayoty) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: Updated according to both feedback.

**Review by soltysh — APPROVED** [review]
/lgtm

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:1056]: nit: this text doesn't make it completely clear what happens if you have multiple replicas of k-c-m and, perhaps by accident, they have different local timezones.

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:1045]: As this is behind `CronJobsScheduledAnnotation`, which will be beta:
```suggestion
### batch.kubernetes.io/cronjob-scheduled-timestamp (beta) {#batch-kubernetes-io-cronjob-scheduled-timestamp}
```

**Review by sftim — COMMENTED** [review]
Please open a PR to document the `CronJobCreationAnnotation`

/hold
We shouldn't merge this yet. I'd prefer to wait until the code implementation has merged and the feature gate docs at least exist as a placeholder.

**Comment by helayoty** [comment]
> We shouldn't merge this yet. I'd prefer to wait until the code implementation has merged and the feature gate docs at least exist as a placeholder.

It just got merged. https://github.com/kubernetes/kubernetes/pull/118137

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (helayoty) [reply content/en/docs/reference/labels-annotations-taints/_index.md:1056]: We may target this in the GA, @soltysh  wdyt?

Comment (sftim) [reply content/en/docs/reference/labels-annotations-taints/_index.md:1056]: We can _document_ the behavior even for beta. However, leaving the detail out doesn't block a release at beta quality.

Comment (soltysh) [reply content/en/docs/reference/labels-annotations-taints/_index.md:1056]: So based on the already merged implementation we will fallback to UTC, because if the value isn't set we use empty timezone, which so happens resolves to UTC, see https://github.com/kubernetes/kubernetes/blob/ddb201336382808b756a140b7cf2039c6de6f846/pkg/controller/cronjob/utils.go#L219-L227 So I'm fine with Tim's proposal to make it explicit, which is definitely better than my initial approach for leaving it for later.

Comment (sftim) [reply content/en/docs/reference/labels-annotations-taints/_index.md:1056]: That sounds ideal all round.

Comment (soltysh) [reply content/en/docs/reference/labels-annotations-taints/_index.md:1056]: @helayoty this needs update and it's good to go
```suggestion
with a timezone specified, then the timestamp is in that timezone. Otherwise, the timestamp is in UTC.
```

**Comment by tengqm** [comment]
@helayoty Please help take a look at the conflicts.

**Comment by helayoty** [comment]
> @helayoty Please help take a look at the conflicts.

@tengqm , done and ready for review/merge

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:1074]: Is this true of ` batch.kubernetes.io/cronjob-scheduled-timestamp`? I think this text needs to move.

**Review by sftim — CHANGES_REQUESTED** [review]

**Comment by sftim** [comment]
/hold cancel

Please see https://github.com/kubernetes/website/pull/41857#discussion_r1287310720 though.

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (helayoty) [reply content/en/docs/reference/labels-annotations-taints/_index.md:1074]: This is not related to the `batch.kubernetes.io/cronjob-scheduled-timestamp` annotation, but the  'kubectl.kubernetes.io/default-logs-container (deprecated)' one.

**Comment by helayoty** [comment]
> /hold cancel
> 
> Please see [#41857 (comment)](https://github.com/kubernetes/website/pull/41857#discussion_r1287310720) though.

Replied. ready for merge.

**Comment by katcosgrove** [comment]
Hey y'all! It's the docs deadline. Is there anything the docs team can do to help merge this?

@sftim You have requested changes on this PR.

File: content/en/docs/reference/labels-annotations-taints/_index.md
Comment (sftim) [reply content/en/docs/reference/labels-annotations-taints/_index.md:1074]: Have a look at https://deploy-preview-41857--kubernetes-io-vnext-staging.netlify.app/docs/reference/labels-annotations-taints/#batch-kubernetes-io-cronjob-scheduled-timestamp

The last paragraph belongs to https://deploy-preview-41857--kubernetes-io-vnext-staging.netlify.app/docs/reference/labels-annotations-taints/#batch-kubernetes-io-job-completion-index - and you should move these 2 lines to achieve that.

Comment (helayoty) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: Got it. Done.

**Review by Rishit-dagli — COMMENTED** [review]
LGTM on the style guide.

**Review by divya-mohan0209 — COMMENTED** [review]
Approving this for merge. 
/approve

**Review by divya-mohan0209 — COMMENTED** [review]
LGTMing it from the comment by @soltysh 
/lgtm

**Pull request merged**
